nsxt/resource_nsxt_vpc.go

```diff
@@ -0,0 +1,626 @@
+			SchemaType:   "bool",
+			SdkFieldName: "ActivateDefaultDfwRules",
+		},
+	},
+	"private_ips": {
+		Schema: schema.Schema{
+			Type: schema.TypeList,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "array",
+			SdkFieldName: "PrivateIps",
+		},
+	},
+	"default_gateway_path": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validatePolicyPath(),
+			Optional:     true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "DefaultGatewayPath",
+		},
+	},
+	"service_gateway": {
+		Schema: schema.Schema{
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"auto_snat": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "bool",
+							SdkFieldName: "AutoSnat",
+						},
+					},
+					"qos_config": {
+						Schema: schema.Schema{
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &metadata.ExtendedResource{
+								Schema: map[string]*metadata.ExtendedSchema{
+									"ingress_qos_profile_path": {
+										Schema: schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validatePolicyPath(),
+											Optional:     true,
+										},
+										Metadata: metadata.Metadata{
+											SchemaType:   "string",
+											SdkFieldName: "IngressQosProfilePath",
+										},
+									},
+									"egress_qos_profile_path": {
+										Schema: schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validatePolicyPath(),
+											Optional:     true,
+										},
+										Metadata: metadata.Metadata{
+											SchemaType:   "string",
+											SdkFieldName: "EgressQosProfilePath",
+										},
+									},
+								},
+							},
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "struct",
+							SdkFieldName: "QosConfig",
+							ReflectType:  reflect.TypeOf(model.GatewayQosProfileConfig{}),
+						},
+					},
+					"disable": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "bool",
+							SdkFieldName: "Disable",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "ServiceGateway",
+			ReflectType:  reflect.TypeOf(model.ServiceGateway{}),
+		},
+	},
+	"vpc_service_profile": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validatePolicyPath(),
+			Optional:     true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "VpcServiceProfile",
+		},
+	},
+	"load_balancer_vpc_endpoint": {
```

Comment:
Can we flatten this? In the end we have MaxItem=1 and a boolean within.

Reply:
for now the strategy is to follow the api..